custom/icds_reports/reports/ucr.py

```diff
@@ -127,3 +130,225 @@ def get_total_row(self):
+            'partial': columns.pse_days_attended.between(1, 15)
+        }[pse]
+        column &= {
+            'male': columns.sex == 'M',
+            'female': columns.sex == 'F',
+        }[gender]
+        if other is None:
+            return func.count(self.table.c.case_id).filter(column).label("pse_{}_{}".format(pse, gender))
+
+        column &= {
+            'st': columns.caste == 'st',
+            'sc': columns.caste == 'sc',
+            'others': columns.caste.notin_(('st', 'sc')),
+            'disabled': columns.disabled == '1',
+            'minority': columns.minority == '1',
+        }[other]
+        return func.count(self.table.c.case_id).filter(column).label("pse_{}_{}_{}".format(pse, gender, other))
+
+    def _columns(self, total_row=False):
+        columns = (
+            self._column_helper("16_days", "male", "st"),
+            self._column_helper("16_days", "female", "st"),
+            self._column_helper("16_days", "male", "sc"),
+            self._column_helper("16_days", "female", "sc"),
+            self._column_helper("16_days", "male", "others"),
+            self._column_helper("16_days", "female", "others"),
+            self._column_helper("16_days", "male", "disabled"),
+            self._column_helper("16_days", "female", "disabled"),
+            self._column_helper("16_days", "male", "minority"),
+            self._column_helper("16_days", "female", "minority"),
+            self._column_helper("absent", "male"),
+            self._column_helper("absent", "female"),
+            self._column_helper("partial", "male"),
+            self._column_helper("partial", "female"),
+            func.count(self.table.c.case_id).filter(self.table.c.sex == 'M').label("child_count_male"),
+            func.count(self.table.c.case_id).filter(self.table.c.sex == 'F').label("child_count_female"),
+        )
+
+        if not total_row:
+            return (self.table.c.awc_id.label("awc_id"),) + columns
+
+        return columns
+
+    def _get_query_object(self, total_row=False):
+        filters = self.helper.sql_alchemy_filters
+        filter_values = self.helper.sql_alchemy_filter_values
+        query = (
+            self.helper.adapter.session_helper.Session.query(
+                *self._columns(total_row)
+            )
+            .filter(*filters)
+            .params(filter_values)
+        )
+        if not total_row:
+            query = query.group_by(self.table.c.awc_id)
+        return query
+
+
+class MPR5ChildHealth(ChildHealthMonthlyUCR):
+    """Note that this is not the same as the original.
+
+    For the original, when children are 36 - 72 months, we only count food
+    distributed for child if it was "hot", but it's possible that there is
+    app logic that prevents that from ever being the case
+    """
+    def _column_helper(self, thr, gender_or_migrant, other=None):
+        columns = self.table.c
+        column = (columns.thr_eligible == 1)
+        column &= {
```

Comment:
What does the ```&=``` syntax mean?  Is this some sqlalchemy syntactic sugar?

Reply:
It's shorthand for `column = 1 AND other_column = 2`. Here this code is generating an expression to use to return a column in the final SQL query